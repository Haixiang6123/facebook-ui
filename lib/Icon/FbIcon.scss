@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.fb-icon {
  width: 1.4em;
  height: 1.4em;
  fill: currentColor;
  &.fb-icon-spin {
    animation: spin .7s infinite linear;
  }
}
