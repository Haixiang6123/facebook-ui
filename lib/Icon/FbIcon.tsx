import * as React from 'react'
import classes from '../utils/classes'
import './icons/index'
import './FbIcon.scss'

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string
  solid?: boolean
  spin?: boolean
}

const FbIcon: React.FunctionComponent<IconProps> =
  ({className, name, solid, spin, ...restProps}) => {
    const link = `#icon_${name}` + (solid ? '_fill' : '')
    const classNames = classes('fb-icon', className, spin && 'fb-icon-spin')
    return (
      <svg className={classNames} {...restProps}>
        <use xlinkHref={link}/>
      </svg>
    )
}

export default FbIcon
