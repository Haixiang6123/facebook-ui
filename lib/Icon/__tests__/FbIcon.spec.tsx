import * as React from 'react'
import * as renderer from 'react-test-renderer'
import {mount, shallow} from 'enzyme'
import FbIcon from '../FbIcon'

describe('FbIcon', () => {
  it('should render', () => {
    const json = renderer.create(<FbIcon name="video"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('should handle event', () => {
    const handler = jest.fn()
    const component = mount(<FbIcon name="video" onClick={handler}/>)
    component.find('svg').simulate('click')
    expect(handler).toBeCalled()
  })
  it('should receive props solid', () => {
    const solidComponent = shallow(<FbIcon name="shakehands" solid={true}/>)
    const notSolidComponent = shallow(<FbIcon name="shakehands" solid={false}/>)
    expect(solidComponent.find('use').prop('xlinkHref')).toEqual('#icon_shakehands_fill')
    expect(notSolidComponent.find('use').prop('xlinkHref')).toEqual('#icon_shakehands')
  })
  it('should receive props spin', () => {
    const spinComponent = shallow(<FbIcon name="video" spin={true}/>)
    const notSpinComponent = shallow(<FbIcon name="video" spin={false}/>)
    expect(spinComponent.hasClass('fb-icon-spin')).toEqual(true)
    expect(notSpinComponent.hasClass('fb-icon-spin')).toEqual(false)
  })
})
