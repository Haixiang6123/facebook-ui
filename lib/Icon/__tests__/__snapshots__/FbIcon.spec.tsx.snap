// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FbIcon should render 1`] = `
<svg
  className="fb-icon"
>
  <use
    xlinkHref="#icon_video"
  />
</svg>
`;
